instrumentation/jsf/jsf-javax-common/testing/src/main/java/io/opentelemetry/javaagent/instrumentation/jsf/javax/BaseJsfTest.java

```diff
@@ -0,0 +1,325 @@
+import io.opentelemetry.instrumentation.testing.junit.http.AbstractHttpServerUsingTest;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpServerInstrumentationExtension;
+import io.opentelemetry.sdk.testing.assertj.SpanDataAssert;
+import io.opentelemetry.sdk.testing.assertj.TraceAssert;
+import io.opentelemetry.sdk.trace.data.StatusData;
+import io.opentelemetry.semconv.ClientAttributes;
+import io.opentelemetry.semconv.ExceptionAttributes;
+import io.opentelemetry.semconv.HttpAttributes;
+import io.opentelemetry.semconv.NetworkAttributes;
+import io.opentelemetry.semconv.ServerAttributes;
+import io.opentelemetry.semconv.UrlAttributes;
+import io.opentelemetry.semconv.UserAgentAttributes;
+import io.opentelemetry.testing.internal.armeria.common.AggregatedHttpRequest;
+import io.opentelemetry.testing.internal.armeria.common.AggregatedHttpResponse;
+import io.opentelemetry.testing.internal.armeria.common.HttpData;
+import io.opentelemetry.testing.internal.armeria.common.HttpMethod;
+import io.opentelemetry.testing.internal.armeria.common.MediaType;
+import io.opentelemetry.testing.internal.armeria.common.QueryParams;
+import io.opentelemetry.testing.internal.armeria.common.RequestHeaders;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+import javax.servlet.ServletException;
+import org.eclipse.jetty.annotations.AnnotationConfiguration;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.util.resource.Resource;
+import org.eclipse.jetty.webapp.WebAppContext;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.ArgumentsProvider;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+
+public abstract class BaseJsfTest extends AbstractHttpServerUsingTest<Server> {
+  @RegisterExtension
+  public static final InstrumentationExtension testing =
+      HttpServerInstrumentationExtension.forAgent();
+
+  protected abstract String getJsfVersion();
+
+  @BeforeAll
+  protected void setUp() {
+    startServer();
+  }
+
+  @AfterAll
+  protected void cleanUp() {
+    cleanupServer();
+  }
+
+  @Override
+  protected Server setupServer() throws Exception {
+    List<String> configurationClasses = new ArrayList<>();
+    addAll(configurationClasses, WebAppContext.getDefaultConfigurationClasses());
+    configurationClasses.add(AnnotationConfiguration.class.getName());
+
+    WebAppContext webAppContext = new WebAppContext();
+    webAppContext.setContextPath(getContextPath());
+    // set up test application
```

Comment:
It seems you miss below line above this line:
```java
webAppContext.setConfigurationClasses(configurationClasses);
```

Reply:
thanks!